### PR TITLE
Pre authorize annotations for subresources

### PR DESCRIPTION
**Pull request by benbosman**
## References
_Add references/links to any related tickets or PRs. These may include:_
* Link to [initial REST implementation](https://github.com/DSpace/DSpace/pull/2726#issuecomment-628586882)

## Description
When including a subresource, e.g. via embedding, the permissions of the subresources will now be verified based on the @PreAuthorize setting of the findOne() method.  
The default @PreAuthorize on the abstract findOne() method has been set to @PreAuthorize("hasAuthority('ADMIN')").

## Instructions for Reviewers
The ConverterService will now use the @PreAuthorize to determine whether a resource can be embedded. It will automatically replace the `#id` in the annotation with the object's ID (but the syntax should be `#id`)

The ITs were already included in https://github.com/DSpace/DSpace/pull/2726, they still verify that the permissions are enforced on the subresources.

Some PermissionEvaluatorPlugins which were required for https://github.com/DSpace/DSpace/pull/2726 have been removed again because the permissions can now be retrieved from the findOne() method

Since there was a conflict on the permissions of the AuthorizationFeatureRest (sometimes anonymous and sometimes admin), this has been adjusted here as well

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [x] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/master/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR modifies the REST API, I've linked to the REST Contract page (or open PR) related to this change.

### CODE REVIEW

**Comment by crosenbeck** [comment]
@benbosman and @Raf-atmire Thank you for work on this PR. It looks good to me.

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/AuthorizationRestRepositoryIT.java [hunk 1]
```diff
@@ -804,6 +805,11 @@ public void findByObjectInternalServerErrorTest() throws Exception {
      * 
      * @throws Exception
      */
+    // This test currently doesn't work as expected since the AuthorizationFeatureRestRepository#findOne method
+    // is only exposed to admins. Currently we're performing checks on the individual REST objects with its findOne
+    // Permission constraints, which is ADMIN in this case. Seeing as we're trying to retrieve it with a normal
+    // EPerson token in the second test, this will fail.
+    @Ignore
```
Comment (abollini) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/AuthorizationRestRepositoryIT.java:812]: this should be possible now
see https://github.com/DSpace/DSpace/pull/2778/files#diff-311a7e07c257d9c869f4de1347f5f761R51

**Review by abollini — APPROVED** [review]
code looks good to me. Just a minor comment about an additional test that can be re-enabled now (I guess)

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/AuthorizationRestRepositoryIT.java (hunk 1 above)
Comment (benbosman) [reply dspace-server-webapp/src/test/java/org/dspace/app/rest/AuthorizationRestRepositoryIT.java]: Removing the `@Ignore` does indeed work now

**Pull request merged**